src/foam/dao/EasyDAO.js

```diff
@@ -279,8 +282,31 @@ return delegate;
         .setOf(getOf())
         .build();
       }
-      if ( getJournalType().equals(JournalType.SINGLE_JOURNAL) )
+      if ( getJournalType().equals(JournalType.SINGLE_JOURNAL) ) {
         return new foam.dao.java.JDAO(getX(), getOf(), getJournalName());
+      }
+      else if ( getJournalType().equals(JournalType.SHARED_JOURNAL) ) {
+
+        foam.dao.MDAO myMDAO = new foam.dao.MDAO(getOf());
+        X resourceStorageX = getX();
+        if ( System.getProperty("resource.journals.dir") != null ) {
+          resourceStorageX = getX().put(foam.nanos.fs.Storage.class,
+              new ResourceStorage(System.getProperty("resource.journals.dir")));
+        }
+        new foam.dao.FileJournal.Builder(resourceStorageX)
+          .setFilename(getSourceJournalName() + ".0")
+          .build().replay(getX(), myMDAO);
+        SharedJournalFactorySingleton sharedJournalFactory =
+          (SharedJournalFactorySingleton) getX().get("sharedJournalFactory");
+        foam.dao.Journal jrl =
+          sharedJournalFactory.getJournal(
+            getJournalName(), getNSpec().getName(), myMDAO);
+        return new foam.dao.java.JDAO.Builder(getX())
+          .setDelegate(myMDAO)
+          .setOf(getOf())
+          .setJournal(jrl)
+          .build();
+      }
```

Comment:
Another PR (or this one if requested) could implement a facade (as per the comment in JDAO) to contain this code, and the similar code for SINGLE_JOURNAL that's inside JDAO right now.